JSONAPI.TodoMVC.API/Startup.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Data.Entity;
+using System.Reflection;
+using System.Web;
+using System.Web.Http;
+using Autofac;
+using Autofac.Integration.WebApi;
+using JSONAPI.ActionFilters;
+using JSONAPI.Core;
+using JSONAPI.EntityFramework.ActionFilters;
+using JSONAPI.Json;
+using JSONAPI.TodoMVC.API.Controllers;
+using JSONAPI.TodoMVC.API.Models;
+using Microsoft.Owin;
+using Owin;
+
+namespace JSONAPI.TodoMVC.API
+{
+    public class Startup
+    {
+        private const string DbContextKey = "TodoMVC.DbContext";
+
+        private readonly Func<IOwinContext, TodoMvcContext> _dbContextFactory;
+
+        public Startup()
+            : this(context => new TodoMvcContext())
+        {
+            
+        }
+
+        public Startup(Func<IOwinContext, TodoMvcContext> dbContextFactory)
+        {
+            _dbContextFactory = dbContextFactory;
+        }
+
+        public void Configuration(IAppBuilder app)
+        {
+            // Setup db context for use in DI
+            app.Use(async (context, next) =>
+            {
+                TodoMvcContext dbContext = _dbContextFactory(context);
+                context.Set(DbContextKey, dbContext);
+
+                await next();
+
+                dbContext.Dispose();
+            });
+
+            var appContainerBuilder = new ContainerBuilder();
+            appContainerBuilder.Register(ctx => HttpContext.Current.GetOwinContext()).As<IOwinContext>();
+            appContainerBuilder.Register(c =>
+            {
+                return c.Resolve<IOwinContext>().Get<TodoMvcContext>(DbContextKey);
```

Comment:
When a dependency of type TodoMvcContext (the return type of the lambda) is requested, the lambda will be run. How often it runs is determined by the lifetime rules of the registration. I believe by default, it gets run every time a dependency is requested. So every time TodoMvcContext is resolved, a new one will get generated. But I should probably specify that this uses `InstancePerMatchingLifetimeScope()`.